openedx/core/djangoapps/catalog/utils.py

```diff
@@ -276,12 +276,18 @@ def get_fulfillable_course_runs_for_entitlement(entitlement, course_runs):
     2) A user can enroll in
     3) A user can upgrade in
     4) Are published
+    5) Are not enrolled in already for an active session
```

Comment:
Active? That doesn't seem to be true with below.  Just enrolled at all right?

Reply:
This comes from the method of CourseEnrollment.enrollments_for_user, ```
        return cls.objects.filter(user=user, is_active=1).select_related('user')```